### PR TITLE
Added support for the `kotlin.time.Duration` class as built-in

### PR DESCRIPTION
**Pull request by shanshin**

### CODE REVIEW

File: formats/json/commonTest/src/kotlinx/serialization/features/DurationTest.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import kotlin.test.Test
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+
+class DurationTest : JsonTestBase() {
+
+    @Serializable
+    data class DurationHolder(val duration: Duration)
```
Comment (qwwdfsad) [inline formats/json/commonTest/src/kotlinx/serialization/features/DurationTest.kt:18]: For now, you can test `Duration.serializer()` explicitly without a holder

File: core/commonMain/src/kotlinx/serialization/builtins/BuiltinSerializers.kt [hunk 2]
```diff
@@ -217,3 +218,9 @@ public fun UByte.Companion.serializer(): KSerializer<UByte> = UByteSerializer
 @ExperimentalSerializationApi
 @ExperimentalUnsignedTypes
 public fun UShort.Companion.serializer(): KSerializer<UShort> = UShortSerializer
+
+/**
+ * Returns serializer for [Duration].
+ * By default, it is serialized as a string that represents a duration in ISO-8601 format.
+ */
+public fun Duration.Companion.serializer(): KSerializer<Duration> = DurationSerializer
```
Comment (sandwwraith) [inline core/commonMain/src/kotlinx/serialization/builtins/BuiltinSerializers.kt:226]: Shouldn't it be experimental?

File: formats/json/commonTest/src/kotlinx/serialization/features/DurationTest.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonTestBase
+import kotlin.test.Test
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+
+class DurationTest : JsonTestBase() {
```
Comment (sandwwraith) [inline formats/json/commonTest/src/kotlinx/serialization/features/DurationTest.kt:15]: I think it makes sense to have two tests: one commented that have `DurationHolder` and checks that plugin correctly bounds serializer (existing now), and the other, uncommented, that explicitly uses `Duration.serializer()` and verifies that actual serializer works correctly.

**Review by sandwwraith — CHANGES_REQUESTED** [review]

File: core/commonMain/src/kotlinx/serialization/builtins/BuiltinSerializers.kt (hunk 2 above)
Comment (shanshin) [reply core/commonMain/src/kotlinx/serialization/builtins/BuiltinSerializers.kt:226]: @qwwdfsad

File: formats/json/commonTest/src/kotlinx/serialization/features/DurationTest.kt (hunk 3 above)
Comment (shanshin) [reply formats/json/commonTest/src/kotlinx/serialization/features/DurationTest.kt:15]: Added separated test on `DurationSerializer` to `BasicTypesSerializationTest` but left these test.
Also added extra check to commented test in `SerializersLookupTest`.

**Review by sandwwraith — APPROVED** [review]

File: core/commonMain/src/kotlinx/serialization/builtins/BuiltinSerializers.kt (hunk 2 above)
Comment (qwwdfsad) [reply core/commonMain/src/kotlinx/serialization/builtins/BuiltinSerializers.kt:228]: It cannot be experimental at the moment it has built-in plugin support -- users won't get experimenta warnings on `@Serializable class D(val d: Duration)`, rendering our own experimental warning here useless

**Review by qwwdfsad — APPROVED** [review]

**Pull request merged**